tests/test_python3.py

```diff
@@ -63,3 +66,24 @@ def test_python3_with_docker():
     assert output_file.readline() == "[[ 58  64]\n"
     assert output_file.readline() == " [139 154]]"
     output_file.close()
+
+
+@pytest.mark.integration
+def test_python3_with_public_docker():
+    """
+    Tests using a public docker image with the write test script
+    """
+
+    test_dir = "./temp_test_python3/with_public_docker"
+    os.makedirs(f"{test_dir}", exist_ok=True)
+    toolchest.python3(
+        tool_args="./input/example.fastq",
+        script="s3://toolchest-integration-tests/write_test.py",
+        inputs="s3://toolchest-integration-tests/example.fastq",
+        output_path=f"{test_dir}/",
+        custom_docker_image_id="python:alpine3.16",
+    )
+
+    output_file = open(f"{test_dir}/output.txt", "r")
+    assert output_file.readline() == "Success"
+    output_file.close()
```

Comment:
This tests that the command succeeds, but it doesn't test that it actually executed in the image that's specified nor that this works for an image with a registry URL.

I would be okay with merging this as-is because it it's a net improvement, but imo this isn't testing the feature.

Reply:
It is testing the feature (although maybe not a clearly as a custom image). If the image doesn't have python3 command accessible or failed to download the tests fails. If the image exists locally it would upload which is a downside but that won't be the case currently.

Reply:
Yeah, I see that this test would fail if a Docker error is thrown (on client or on worker node) or if the image `python:alpine3.16` doesn't have `python3`. What I see missing as missing is that this also works on the base instance – nothing about this test is tied to the custom Docker image. I believe that it works now, but as a regression test I think it's suboptimal.

For example, with Lug, we had a very similar test for custom images (function runs and returns value X). However, the "function runs and returns value X" wasn't unique to the custom image; it succeeded even if the command wasn't rerouted to the custom image. That's why we didn't realize it wasn't functioning as intended for a while: the test passed, but if a component failed silently there was no test failure.

I see the same thing here: yes, this hits the path that a custom docker image upload path would hit. But this integration test isn't testing that:
- the Docker image uploaded
- the command is executing in that Docker image

> If the image exists locally it would upload which is a downside but that won't be the case currently.

I have that image locally already!